indico/modules/events/models/events.py

```diff
@@ -683,16 +687,20 @@ def get_relative_event_ids(self):
                                                               range_=(None, None)).label('last'),
                             db.func.lag(Event.id).over(order_by=(Event.start_dt, Event.id)).label('prev'),
                             db.func.lead(Event.id).over(order_by=(Event.start_dt, Event.id)).label('next')])
-                    .where((Event.category_id == self.category_id) & ~Event.is_deleted)
+                    .where((Event.category_id == self.category_id) & ~Event.is_deleted &
+                           (Event.visibility.is_(None) | (Event.visibility != 0)))
                     .alias())
         rv = (db.session.query(subquery.c.first, subquery.c.last, subquery.c.prev, subquery.c.next)
               .filter(subquery.c.id == self.id)
-              .one()
-              ._asdict())
-        if rv['first'] == self.id:
-            rv['first'] = None
-        if rv['last'] == self.id:
-            rv['last'] = None
+              .first())
+        if not rv:
```

Comment:
I think you can avoid this case altogether by not applying the visibility filter in case, i.e. `(Event.visibility.is_(None) | (Event.visibility != 0) | (Event.id == self.id))`

Reply:
Looks much cleaner: I'm going to test it, tnx.